### PR TITLE
net/bluetooth: LE Connection and disconnection complete event

### PR DESCRIPTION
**Pull request by amitpurwar**
LE Connection and disconnection complete event

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [fdc2e0b0a56b34aff6cd8b03d4d2e20db62cba25] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fdc2e0b0a56b34aff6cd8b03d4d2e20db62cba25] - Code Rule Check Result:
os/include/tinyara/bluetooth/bt_hci.h:332: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/bluetooth/bt_hci.h:332: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/tinyara/bluetooth/bt_hci.h:335: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/bluetooth/bt_hci.h:335: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/tinyara/bluetooth/bt_hci.h:337: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/bluetooth/bt_hci.h:337: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/tinyara/bluetooth/bt_hci.h:339: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/bluetooth/bt_hci.h:339: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/tinyara/bluetooth/bt_hci.h:341: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/bluetooth/bt_hci.h:341: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/tinyara/bluetooth/bt_hci.h:343: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/bluetooth/bt_hci.h:343: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/tinyara/bluetooth/bt_hci.h:345: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/bluetooth/bt_hci.h:345: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/tinyara/bluetooth/bt_hci.h:347: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/bluetooth/bt_hci.h:347: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [f5c5e564ae24fda3d4bffde8049747f0d304d1ec] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f5c5e564ae24fda3d4bffde8049747f0d304d1ec] - Code Rule Check OK.

File: os/net/bluetooth/bt_hcicore.h [hunk 1]
```diff
@@ -450,4 +473,14 @@ int hci_le_create_conn(FAR const bt_addr_le_t *addr);
 
 void cmd_queue_init(void);
 
+void hci_disconn_complete_internal(FAR struct bt_buf_s *buf);
+
```
Comment (NilabhShant) [inline os/net/bluetooth/bt_hcicore.h]: Please remove these extra lines.

File: os/net/bluetooth/bt_util.h [hunk 2]
```diff
@@ -0,0 +1,118 @@
+/****************************************************************************
+ * include/tinyara/bluetooth/bt_null.h
```
Comment (NilabhShant) [inline os/net/bluetooth/bt_util.h]: Please correct the the path as well as file name.

File: os/net/bluetooth/bt_util.h [hunk 3]
```diff
@@ -0,0 +1,118 @@
+/****************************************************************************
```
Comment (NilabhShant) [inline os/net/bluetooth/bt_util.h]: Please add Samsung license header .

File: os/net/bluetooth/bt_util.h [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+ *   ENABLED:    _YYYY,    <--- note comma!
+ *   DISABLED:   _XXXX
+ */
+#define _XXXX1 _YYYY,
+
+/* Then we append an extra argument to fool the gcc preprocessor into
+ * accepting it as a varargs macro.
+ *                         arg1   arg2  arg3
+ *   ENABLED:   Z_IS_ENABLED3(_YYYY,    1,    0)
+ *   DISABLED   Z_IS_ENABLED3(_XXXX 1,  0)
```
Comment (NilabhShant) [inline os/net/bluetooth/bt_util.h]: add comma.

 DISABLED   Z_IS_ENABLED3(_XXXX, 1, 0)

File: os/net/bluetooth/bt_util.h (hunk 3 above)
Comment (amitpurwar) [reply os/net/bluetooth/bt_util.h]: Added the Samsung License

File: os/net/bluetooth/bt_util.h (hunk 2 above)
Comment (amitpurwar) [reply os/net/bluetooth/bt_util.h]: Corrected the file name

File: os/net/bluetooth/bt_util.h (hunk 4 above)
Comment (amitpurwar) [reply os/net/bluetooth/bt_util.h]: comma was added for below 
#define _XXXX1 _YYYY,

comma is not required for _XXXX

File: os/net/bluetooth/bt_hcicore.h (hunk 1 above)
Comment (amitpurwar) [reply os/net/bluetooth/bt_hcicore.h]: The blank line is required for readability of the multiple functions

**Comment by seinfra** [comment]
Target : [b11cd11b7c5a19547a264289aba04e7e5c2a46ec] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b11cd11b7c5a19547a264289aba04e7e5c2a46ec] - Code Rule Check OK.

**Pull request merged**